lib/DiscoveryStreamFeed.jsm

```diff
@@ -163,25 +163,84 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     }
   }
 
+  /**
+   * buildFeedPromise - Creates a promise for a single unique feed
+   *                    and populates newFeeds with the result.
+   * @param {Object} Has both newFeedsPromises (Array) and newFeeds (Object)
+   * @param {Boolean} isStartup We have different cache handling for startup.
+   * @returns {Function} A for each function, we return it so we can
```

Comment:
I would instead mention what kind of data it transforms: `A function that works over components with valid 'feed' fields` or something like that.
Similar for the reduce cb `reduceFeedComponents`.